FirebaseStorageSwift/Sources/StorageReference.swift

```diff
@@ -0,0 +1,405 @@
+   * in gs://bucket/path/to/object.txt, the name of the object would be: 'object.txt'
+   */
+  @objc public let name: String
+
+  /**
+   * Creates a new StorageReference pointing to the root object.
+   * @return A new StorageReference pointing to the root object.
+   */
+  @objc open func root() -> StorageReference {
+    return StorageReference(impl: impl.root(), storage: storage)
+  }
+
+  /**
+   * Creates a new StorageReference pointing to the parent of the current reference
+   * or nil if this instance references the root location.
+   * For example:
+   *   path = foo/bar/baz   parent = foo/bar
+   *   path = foo           parent = (root)
+   *   path = (root)        parent = nil
+   * @return A new StorageReference pointing to the parent of the current reference.
+   */
+  @objc open func parent() -> StorageReference? {
+    guard let parent = impl.parent() else {
+      return nil
+    }
+    return StorageReference(impl: parent, storage: storage)
+  }
+
+  /**
+   * Creates a new StorageReference pointing to a child object of the current reference.
+   *   path = foo      child = bar    newPath = foo/bar
+   *   path = foo/bar  child = baz    ntask.impl.snapshotwPath = foo/bar/baz
+   * All leading and trailing slashes will be removed, and consecutive slashes will be
+   * compressed to single slashes. For example:
+   *   child = /foo/bar     newPath = foo/bar
+   *   child = foo/bar/     newPath = foo/bar
+   *   child = foo///bar    newPath = foo/bar
+   * @param path Path to append to the current path.
+   * @return A new StorageReference pointing to a child location of the current reference.
+   */
+  @objc(child:) open func child(_ path: String) -> StorageReference {
+    return StorageReference(impl: impl.child(path), storage: storage)
+  }
+
+  // MARK: - Uploads
+
+  /**
+   * Asynchronously uploads data to the currently specified StorageReference,
+   * without additional metadata.
+   * This is not recommended for large files, and one should instead upload a file from disk.
+   * @param uploadData The Data to upload.
+   * @param metadata StorageMetadata containing additional information (MIME type, etc.)
+   * about the object being uploaded.
+   * @return An instance of StorageUploadTask, which can be used to monitor or manage the upload.
+   */
+  @objc(putData:metadata:) @discardableResult
+  open func putData(_ uploadData: Data, metadata: StorageMetadata? = nil) -> StorageUploadTask {
+    return StorageUploadTask(impl.put(uploadData, metadata: metadata?.impl))
+  }
+
+  /**
+   * Asynchronously uploads data to the currently specified StorageReference.
+   * This is not recommended for large files, and one should instead upload a file from disk.
+   * @param uploadData The Data to upload.
+   * @return An instance of StorageUploadTask, which can be used to monitor or manage the upload.
+   */
+  @objc(putData:) @discardableResult open func __putData(_ uploadData: Data) -> StorageUploadTask {
```

Comment:
will the style script let give each attribute it's own line?
```suggestion
  @objc(putData:)
  @discardableResult
  open func __putData(_ uploadData: Data) -> StorageUploadTask {
```
Personally, I think it'd be more readable but is not a big deal